frame/tx-pause/src/tests.rs

```diff
@@ -16,3 +16,173 @@
+
+#[test]
+fn can_set_arbitrary_pause() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(TxPause::pause_call(
+			Origin::signed(mock::PauseOrigin::get()),
+			b"SomePallet".to_vec().try_into().unwrap(),
+			b"some_function".to_vec().try_into().unwrap(),
+		));
+	});
+}
+
+#[test]
+fn can_pause_system_call() {
+	new_test_ext().execute_with(|| {
+		let call = Call::System(frame_system::Call::remark { remark: vec![] });
+
+		assert_ok!(TxPause::pause_call(
+			Origin::signed(mock::PauseOrigin::get()),
+			b"System".to_vec().try_into().unwrap(),
+			b"remark".to_vec().try_into().unwrap(),
+		));
+
+		assert_err!(
+			call.clone().dispatch(Origin::signed(0)),
+			frame_system::Error::<Test>::CallFiltered
+		);
+	});
+}
+
+#[test]
+fn can_pause_specific_call() {
+	new_test_ext().execute_with(|| {
+		let call_paused = Call::Balances(pallet_balances::Call::transfer { dest: 1, value: 1 });
+		let call_not_paused =
+			Call::Balances(pallet_balances::Call::transfer_keep_alive { dest: 1, value: 1 });
+
+		assert_ok!(TxPause::pause_call(
+			Origin::signed(mock::PauseOrigin::get()),
+			b"Balances".to_vec().try_into().unwrap(),
+			b"transfer".to_vec().try_into().unwrap(),
+		));
+
+		assert_err!(
+			call_paused.clone().dispatch(Origin::signed(0)),
+			frame_system::Error::<Test>::CallFiltered
+		);
+		assert_ok!(call_not_paused.clone().dispatch(Origin::signed(0)));
+	});
+}
+
+#[test]
+fn can_unpause_specific_call() {
+	new_test_ext().execute_with(|| {
+		let call_paused = Call::Balances(pallet_balances::Call::transfer { dest: 1, value: 1 });
+
+		assert_ok!(TxPause::pause_call(
+			Origin::signed(mock::PauseOrigin::get()),
+			b"Balances".to_vec().try_into().unwrap(),
+			b"transfer".to_vec().try_into().unwrap(),
+		));
+		assert_err!(
+			call_paused.clone().dispatch(Origin::signed(0)),
+			frame_system::Error::<Test>::CallFiltered
+		);
+
+		assert_ok!(TxPause::unpause_call(
+			Origin::signed(mock::UnpauseOrigin::get()),
+			b"Balances".to_vec().try_into().unwrap(),
+			b"transfer".to_vec().try_into().unwrap(),
+		));
+		assert_ok!(call_paused.clone().dispatch(Origin::signed(0)));
+	});
+}
+
+// GENERAL FAIL/NEGATIVE TESTS ---------------------
+
+#[test]
+fn fails_to_pause_self() {
+	new_test_ext().execute_with(|| {
+		let within_bound_name =
+			(0u8..(MaxNameLen::get() - 1).try_into().unwrap()).collect::<Vec<_>>();
+
+		assert_noop!(
+			TxPause::pause_call(
+				Origin::signed(mock::PauseOrigin::get()),
+				b"TxPause".to_vec().try_into().unwrap(),
+				within_bound_name.try_into().expect("Function name under MaxNameLen")
+			),
+			Error::<Test>::IsUnpausable
+		);
+	});
+}
+
+#[test]
+fn fails_to_pause_unpausable_pallet() {
+	new_test_ext().execute_with(|| {
+		let within_bound_name =
+			(0u8..(MaxNameLen::get() - 1).try_into().unwrap()).collect::<Vec<_>>();
+
+		assert_noop!(
+			TxPause::pause_call(
+				Origin::signed(mock::PauseOrigin::get()),
+				b"UnpausablePallet".to_vec().try_into().unwrap(),
```

Comment:
```suggestion
				b"UnpausablePallet".to_vec().try_into().unwrap(),
```

You can make a function in the tests `fn name(bytes: &[u8]) -> BoundedVec` and then do `name(b"UnpausablePallet")` and everywhere else this pattern of `.to_vec().try_into().unwrap()` is used.